### PR TITLE
Few front end changes

### PR DESCRIPTION
**Pull request by ahsu1230**
- Long library breaks hover card
  https://app.asana.com/0/13310647066703/19068363415722
- hex color regex is created once

@andrewconner @2is10

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/profile/profile.styl [hunk 1]
```diff
@@ -161,7 +161,6 @@ input.profile-input-input
 .profile-input-placeholder
 input.profile-input-input
   border-bottom: 1px solid #d2dae3
-  padding-right: 58px
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/profile/profile.styl:164]: fixes covering over user's first name in user settings page

File: kifi-backend/modules/common/app/com/keepit/model/Library.scala [hunk 2]
```diff
@@ -214,12 +214,12 @@ object BasicLibraryStatistics {
 }
 
 class HexColor private (val hex: String) {
-  require(hex.matches(HexColor.regexStr), "hex color must be in format '#aaaaaa'")
+  require(HexColor.regex.findFirstIn(hex).isDefined, "hex color must be in format '#aaaaaa'")
```
Comment (ahsu1230) [inline kifi-backend/modules/common/app/com/keepit/model/Library.scala:217]: thanks for the suggestions on this one! @andrewconner @2is10

**Comment by andrewconner** [comment]
Thanks

File: kifi-backend/modules/shoebox/angular/src/libraries/libraryMiniCard.tpl.html [hunk 3]
```diff
@@ -17,7 +17,7 @@
   </div>
 
   <div class="kf-library-mini-card-body">
-    <div class="kf-library-mini-card-name" ng-click="view()">{{library.name}}</div>
+    <div kf-ellipsis full-text="library.name" max-num-lines=1 class="kf-library-mini-card-name" ng-click="view()"></div>
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/libraries/libraryMiniCard.tpl.html:20]: One-line ellipsis is simpler and faster with just CSS.

``` css
white-space: nowrap;
overflow: hidden;
text-overflow: ellipsis;
```

**Comment by 2is10** [comment]
thanks!

File: kifi-backend/modules/shoebox/angular/src/libraries/libraryMiniCard.tpl.html (hunk 3 above)
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/libraries/libraryMiniCard.tpl.html:20]: arggh you're right #facepalm
thanks lol

**Pull request merged**